### PR TITLE
Access CamelCase dictionary keys using snake_case

### PR DESCRIPTION
**Pull request by era**
It fix https://github.com/vintasoftware/tapioca-wrapper/issues/27. Is this what you were thinking? Should I change anything?

Regards,

### CODE REVIEW

**Comment by vu3jej** [comment]
@era  Correct me if I am wrong, but isn't it the opposite of what you have done? I thought it was about converting camel case to snake case. Thanks!

**Comment by era** [comment]
Hello @vu3jej,

 It is exactly what it does, see the test at my commit: https://github.com/era/tapioca-wrapper/commit/d31d62171e2dccd915aca743076e9964b2460cf5. You access using response().camel_case() and  the tapiocaClient changes to camelCase.

I didn't change the key directly in the dict to avoid break everyone that already uses it as camelCase. So both ways work now.

File: tests/test_tapioca.py [hunk 1]
```diff
@@ -52,6 +52,28 @@ def test_in_operator(self):
...
+
+        responses.add(responses.GET, next_url,
+                      body='{"data": [{"key_snake": "value", "camelCase": "data in camel case"}], "paging": {"next": ""}}',
+                      status=200,
+                      content_type='application/json')
+
+        response = self.wrapper.test().get()
+
+
+        for item in response().pages():
```
Comment (filipeximenes) [inline tests/test_tapioca.py]: Why do you need paging for this test?

**Comment by filipeximenes** [comment]
This is working properly but i think a "fallback" approach is safer, cleaner and more readable. Eg:
```
def _get_client_from_name_or_fallback(self, name):
	client = self._get_client_from_name(name)
	if client:
		return client
	
	camel_case_name = self._to_camel_case(name)
	client = self._get_client_from_name(self, camel_case_name)
	if client:
		return client
```
Also... this works with lowerCamelCase, should we support NormalCamelCase?

**Comment by filipeximenes** [comment]
One more thing: this should be documented. Probably not on the quickstart, we need a new section to properly document features/methods. It may be called "TapiocaClient". 

Documentation is very important and we are really behind on it... lots of open documentation issues. I'd appreciate some help with this since I'm little short on time :)

File: tests/test_tapioca.py (hunk 1 above)
Comment (era) [inline tests/test_tapioca.py]: It doesn't, but I was worried with this case too, since I change it in a recent past. Should I remove?

**Comment by era** [comment]
@filipeximenes Hello, I'll do the changes and the documentation but probably not on the next few days. I'm moving and right now I'm living in a hostel while I look for a house.

Regards :)

**Comment by filipeximenes** [comment]
No worries, take your time.
Thanks!

**Comment by era** [comment]
Hi,

I just added the fallback method, I also added support to NormalCamelCase . Is everything okay to merge?

Regards,

File: tapioca/tapioca.py [hunk 2]
```diff
@@ -59,6 +59,32 @@ def __call__(self, *args, **kwargs):
...
+    '''
+    def _to_camel_case(self, name):
+        if isinstance(name, int):
+            return name
+        components = name.split('_')
+        return components[0] + "".join(x.title() for x in components[1:])
+
+    def _get_client_from_name_or_fallback(self, name):
+        client = self._get_client_from_name(name)
+        if client != None:
```
Comment (filipeximenes) [inline tapioca/tapioca.py]: use ``is`` to compare to ``None``

File: tests/test_tapioca.py (hunk 1 above)
Comment (filipeximenes) [inline tests/test_tapioca.py]: Can you simplify this test?
The more unitary it is, the best

**Comment by filipeximenes** [comment]
Thanks! It needs documentation before merging, otherwise we will keep getting behind on documentation :)

**Comment by era** [comment]
I'll do the documentation and fix the test :).

Also, about the test, how do I get the property without the pagination? Should response().data().snake_key().data() work?

**Comment by filipeximenes** [comment]
```response.snake_key().data()```
or 
```response.data()['snake_key']```

**Comment by era** [comment]
@filipeximenes Can you help me with this test? I only removed the pagination, but now the tests are failing :(

**Comment by aericson** [comment]
`response.data.camel_case().data()`
or
`response.data['camel_case']().data()`

**Comment by aericson** [comment]
Also, be careful to keep code compatible with PEP8.

**Comment by era** [comment]
@aericson Thanks! It works! :) Now I'll write the documentation, thank you very much.

**Comment by era** [comment]
Guys,

Could you please check if the documentation is in the right place?

Thanks :)

**Comment by filipeximenes** [comment]
Nice, this works. I want to have a separate doc file for describing stuff more int depth but this works for now.

**Comment by filipeximenes** [comment]
@era hey I thought I had merged this!
I just ported docs to use rst and because of this it now has conflicts.
Could you re-issue this?
I'm very sorry.

**Comment by era** [comment]
@filipeximenes No problem :), I merged in my branch, is it okay for you? If is not I can cherry pick my commits in a new branch and make a new pull request.

**Comment by filipeximenes** [comment]
That's ok. 
Did you fix the tests  to remove the unneeded complexity?

**Comment by era** [comment]
Yes :), now there is no pagination :)

**Comment by filipeximenes** [comment]
Awesome! Thanks!

**Pull request merged**